contxt/cli/commands/iot.py

```diff
@@ -113,3 +115,193 @@ def data(
     print(f"Writing data to {output}...")
     flat_data = [{"timestamp": k, **v} for k, v in data.items()]
     Serializer.to_csv(flat_data, output)
+
+
+@iot.command()
+@click.argument("feed_key", type=str)
+@click.argument("worksheet_file", type=click.File("r"))
+@click.pass_obj
+def ingest_worksheet(clients: Clients, feed_key, worksheet_file) -> None:
+    """Field Worksheet Ingestor"""
+    feed = clients.iot.get_feed_with_key(feed_key)
+    if not feed:
+        print(f"Feed with key {feed_key} does not exist")
+        return
+    groupings_by_label = {}
+    groupings = clients.iot.get_field_groupings_for_facility(feed.facility_id)
+    for grouping in groupings:
+        groupings_by_label[grouping.label] = grouping
+
+    field_objs = []
+    fields_for_feed = clients.iot.get_fields_for_feed(feed_id=feed.id)
+    fields_descriptors_for_feed = [f.field_descriptor for f in fields_for_feed]
+
+    desired_groupings_by_field = {}
+    with worksheet_file as f:
+
+        headers = [
+            "Field Descriptor",
+            "Label",
+            "Data Type",
+            "Units",
+            "Scalar",
+            "Cumulative Value?",
+            "equipment group",
+            "Facility Main?",
+        ]
+        reader = csv.DictReader(f, fieldnames=headers)
+
+        # skip the header
+        next(reader)
+
+        # create a bunch of field objects
+        for row in reader:
+            if row["Label"].replace(" ", "").startswith("-"):
```

Comment:
why?

Reply:
I don't know. This was custom parsing logic that was copypasta from the original PR.
@jrcrawfo Do you know why we would need this? Do we need this?

Reply:
i've remove this